src/Spork/Batch/BatchJob.php

```diff
@@ -93,7 +93,12 @@ public function __invoke()
 
         $results = array();
         foreach ($forks as $fork) {
-            $results = array_merge($results, $fork->getResult());
+            $exitStatus = $fork->getExitStatus();
+            if (0 !== $exitStatus) {
+                // Propagate erroneous state
+                exit($exitStatus);
```

Comment:
The state is indeed propagated but what happens if a child fork throws an exception? Would be very nice to know what exactly went wrong. Or do child forks in that case print the message to stderr?